sbi/neural_nets/embedding_nets/resnet.py

```diff
@@ -0,0 +1,321 @@
+from functools import partial
+from typing import Callable, Type
+
+import torch
+import torch.nn as nn
+
+
+def zero_pad(x: torch.tensor, c_out: int) -> torch.tensor:
+    """
+    Add additional channels to the input tensor by filling it with zeros.
+
+    Args:
+        x:      Input tensor
+        c_out:  Number of output channels
+
+    Returns:
+        x_aug:      Input tensor with additional channels filled with zeros
+    """
+
+    # Check if it is a image like iput
+    assert len(x.shape) == 4
+
+    # Check if the number of channels is increased
+    assert c_out >= x.shape[1]
+
+    c_in = x.shape[1]
+    a = torch.zeros(x.shape[0], c_out - c_in, x.shape[2], x.shape[3], device=x.device)
+
+    x_aug = torch.cat([x, a], dim=1)
+
+    return x_aug
+
+
+def construct_simple_conv_net(
+    c_in: int,
+    c_out: int,
+    c_hidden: int = 32,
+    activation: Type[nn.Module] = nn.ReLU,
+    downsample: int = False,
+) -> nn.Module:
+    """
+    Default implementation for a simple convolutional network with 3 layers.
+
+    Args:
+        c_in:       Number of input channels
+        c_out:      Number of output channels
+        c_hidden:   Number of hidden channels
+        activation: Constructor for a activation function
+        downsample: Apply a strided convolution to halve
+                    the height and width of the input
+
+    Returns:
+        layers:     A sequential container with the convolutional layers
+    """
+
+    # Define the convolutional layers
+    stride = 2 if downsample else 1
+    layers = nn.Sequential(
+        nn.Conv2d(c_in, c_hidden, kernel_size=3, stride=stride, padding=1),
+        activation(),
+        nn.Conv2d(c_hidden, c_hidden, kernel_size=3, stride=1, padding=1),
+        activation(),
+        nn.Conv2d(c_hidden, c_out, kernel_size=3, stride=1, padding=1),
+    )
+
+    return layers
+
+
+class ResidualBLockConv(nn.Module):
+    def __init__(
+        self,
+        c_in: int,
+        c_out: int | None = None,
```

Comment:
Our current convention is to use the "older" typing concentions: `Union, Optional, Dict, List` ... . So would be good to stick to it throughout.